src/snowflake/snowpark/functions.py

```diff
@@ -684,6 +684,36 @@ def bitshiftleft(
     return call_builtin("bitshiftleft", c, n, _emit_ast=_emit_ast)
 
 
+@publicapi
+def bitshiftright_unsigned(
+    to_shift_column: ColumnOrName, n: Union[Column, int], _emit_ast: bool = True
+) -> Column:
+    """Returns the bitwise negation of a numeric expression.
+
+    Example:
+        >>> df = session.createDataFrame([(-1999)], ['a'])
+        >>> df.select(bitshiftright_unsigned('a', 1)).collect()[0][0]
+        9223372036854774808
+
+        >>> df = session.createDataFrame([(42)], ['a'])
+        >>> df.select(bitshiftright_unsigned('a', 1)).collect()[0][0]
+        21
+
+        >>> df = session.createDataFrame([(-21)], ['a'])
+        >>> df.select(bitshiftright_unsigned('a', 1)).collect()[0][0]
+        9223372036854775797
+    """
+    c = _to_col_if_str(to_shift_column, "bitshiftright_unsigned")
+    max_bit = bitshiftleft(lit(1, _emit_ast=False), 64, _emit_ast=False)
+    unsigned_c = iff(
+        c < 0,
+        bitshiftright(c + max_bit, n, _emit_ast=False),
+        bitshiftright(c, n, _emit_ast=False),
+        _emit_ast=False,
+    )
+    return call_builtin("bitand", unsigned_c, max_bit - 1, _emit_ast=_emit_ast)
```

Comment:
Since the arguments to the public API and the internal `call_builtin` method are different the AST needs to be constructed manually via `build_builtin_fn_apply` and assigned to the `_ast` field of the `Column` instance returned by `call_builtin` before returning. See `functions.bround` for a relevant example.

An expectation test in `tests/ast/functions2.test` would also be very helpful.